From 2354498d6299406e3004c15b1ee2f45f3c9582eb Mon Sep 17 00:00:00 2001
From: Andrew de los Reyes <adlr@chromium.org>
Date: Fri, 7 Jun 2019 10:09:12 -0700
Subject: [PATCH 3/6] EXPERIMENTAL: Remove stream compression when generating a
 stream.

This is a problem when updating a preexisting stream that was Flate
compressed, since the newly generated stream is not compressed.
---
 core/fpdfapi/edit/cpdf_pagecontentgenerator.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/fpdfapi/edit/cpdf_pagecontentgenerator.cpp b/core/fpdfapi/edit/cpdf_pagecontentgenerator.cpp
index d2c8bf204..fd4a2efe9 100644
--- a/core/fpdfapi/edit/cpdf_pagecontentgenerator.cpp
+++ b/core/fpdfapi/edit/cpdf_pagecontentgenerator.cpp
@@ -189,7 +189,7 @@ void CPDF_PageContentGenerator::UpdateContentStreams(
     if (buf->tellp() <= 0)
       page_content_manager.ScheduleRemoveStreamByIndex(stream_index);
     else
-      old_stream->SetDataFromStringstream(buf);
+      old_stream->SetDataFromStringstreamAndRemoveFilter(buf);
   }
 
   page_content_manager.ExecuteScheduledRemovals();
-- 
2.11.0

